File: codesite/python_problems/utils/class_design_utils.py
# def test_input(cls, operations: list[str], arguments: list[list]) -> list[str | int | None]:
def test_input(cls, operations: list[str], arguments: list[list]):
    output = []

    for operation, argument in zip(operations, arguments):
        # Constructor
        if operation == cls.__name__:
            instance = cls(*argument)
            output.append(None)
            continue

        # Method call
        method = getattr(instance, operation)
        result = method(*argument)
        output.append(result)

    return output


# Run tests
def run_tests(
        cls,
        operations_list: list[list[str]],
        arguments_list: list[list[list[int]]],
        show_output: bool = False) -> list[bool]:
    """
    Run a batch of TimeMap tests and compare outputs with expected results.
    If show_output is True, returns [(actual, expected), ...] instead of booleans.
    """
    res = []

    for operations, arguments in zip(operations_list, arguments_list):
        # Prints every test case in the new line.
        print(json.dumps(test_input(cls, operations, arguments)))


# run_tests(MinStack, operations_list, arguments_list)

File: codesite/python_problems/utils/class-design-utils.js
/** 
 * @param {string[]} operations
 * @param {number[][]} args
 * @return {number[][]}
 */
const testInput = (operations, args) => {
   const output = []
   let timeMap;

   for (let index = 0; index < operations.length; index++) {
      const operation = operations[index];
      const argument = args[index];

      if (operation === 'TimeMap') {
         timeMap = new TimeMap(...argument);
         output.push(null);
      } else if (operation === 'set') {
         timeMap.set(...argument);
         output.push(null);
      } else if (operation === 'get') {
         output.push(timeMap.get(...argument));
      }
   };
   return output
}

// Run tests
/**
 * Run a batch of TimeMap tests and compare outputs with expected results.
 * If show_output is True, returns [(actual, expected), ...] instead of booleans.
 * @param {string[][]} operationsList 
 * @param {number[][][]} argumentsList 
 * @returns {boolean}
 */
const runTests = (operationsList, argumentsList, expectedOutputList, showOutput) => {
   const output = [];

   for (let index = 0; index < operationsList.length; index++) {
      const operations = operationsList[index];
      const args = argumentsList[index];
         output.push([testInput(operations, args)])
   }
   return output
}

runTests(operationsList, argumentsList, expectedOutputList)

